eth/gasestimator/gasestimator.go

```diff
@@ -62,6 +62,9 @@ func Estimate(ctx context.Context, call *core.Message, opts *Options, gasCap uin
 	if call.GasLimit >= params.TxGas {
 		hi = call.GasLimit
 	}
+	if hi >= params.MaxTxGas {
```

Comment:
This might break L2 projects whose transactions potentially consume more gas in order to amortize L1 costs.

Reply:
Yeah, you are right!

I will ask L2 forks how they plan to integrate the EIP-7825 first, but generally your changes look good to me.